label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule_config_service.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+import "org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto";
+
+service LabelApplicationRuleConfigService {
+
+  // create label application rule
+  rpc CreateLabelApplicationRule(CreateLabelApplicationRuleRequest) returns (CreateLabelApplicationRuleResponse) {}
+
+  // get label application rule by id
+  rpc GetLabelApplicationRule(GetLabelApplicationRuleRequest) returns (GetLabelApplicationRuleResponse) {}
+
+  // get all label application rules
+  rpc GetLabelApplicationRules(GetLabelApplicationRulesRequest) returns (GetLabelApplicationRulesResponse) {}
+
+  // update label application rule
+  rpc UpdateLabelApplicationRule(UpdateLabelApplicationRuleRequest) returns (UpdateLabelApplicationRuleResponse) {}
+
+  // delete label application rule by id
+  rpc DeleteLabelApplicationRule(DeleteLabelApplicationRuleRequest) returns (DeleteLabelApplicationRuleResponse) {}
+
+}
+
+message CreateLabelApplicationRuleRequest {
+  CreateLabelApplicationRule label_application_rule = 1;
+}
+
+message CreateLabelApplicationRuleResponse {
+  LabelApplicationRule label_application_rule = 1;
+}
+
+message GetLabelApplicationRuleRequest {
+  string id = 1;
+}
+
+message GetLabelApplicationRuleResponse {
+  LabelApplicationRule label_application_rule = 1;
+}
+
+message GetLabelApplicationRulesRequest {
+}
+
+message GetLabelApplicationRulesResponse {
+  repeated LabelApplicationRule llabel_application_rules = 1;
+}
+
+message UpdateLabelApplicationRuleRequest {
+  LabelApplicationRule label_application_rule = 1;
```

Comment:
By separating this out here, we leave room to add read-only fields in the future. For example, imagine adding a creation date field to the rule, which we would not want to support for editing.

e.g.
```proto
message UpdateLabelApplicationRuleRequest {
  string id = 1;
  LabelApplicationRule data = 2;
}
```

Reply:
I didn't fully get this. In config service, we auto populate create and update timestamp. This is not even exposed as part of individual service config.

Reply:
We don't expose these today, create timestamp was just being used as an example of a field we might want defined on a rule, but never exposed for either create or update.

Reply:
Sure. I don't have a problem with this approach. Updated accordingly.